alacritty_terminal/src/tty/windows/blocking.rs

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+//! Code for running a reader/writer on another thread while driving it through `polling`.
+
+use piper::{pipe, Reader, Writer};
+use polling::os::iocp::{CompletionPacket, PollerIocpExt};
+use polling::{Event, PollMode, Poller};
+
```

Comment:
Do we really need this code here? I'd assume some other clients could be interested in it? We also don't have SPDX prefixes in the project in case they match our licenses.

Reply:
I would definitely be willing to extract this into a `polling-utils`-esque crate for other users.

Reply:
Why is a different license necessary here to begin with?

Reply:
Removed the notice. I thought there was an Apache 2.0 license, my bad

Reply:
The crate is apache only, so you should keep the notice if you want. Though, you can just move the part of you want to your utils.

Reply:
As the original code author @notgull can do whatever he wants with the code he wrote himself. This includes inclusion in his own projects under whatever license he wants.

The only difference this would make is that this would give a different license option to people using this code (like ourselves). But since Apache and MIT are compatible licenses this isn't really a major factor.